assets/helpers/user/userReducer.js

```diff
@@ -1,34 +1,76 @@
 // @flow
 
 // ----- Imports ----- //
-import type { Action } from './userActions';
-
+import { emptyInputField, validateEmailAddress } from 'helpers/utilities';
+import {
+  setFirstNameShouldValidate,
+  setLastNameShouldValidate,
+  setEmailShouldValidate,
+  setFullNameShouldValidate,
+  setEmail,
+  setFirstName,
+  setLastName,
+  setFullName,
+  type Action,
+} from './userActions';
 
 // ----- Types ----- //
+export type UserFormFieldAttribute = {
+  value: string,
+  shouldValidate: boolean,
+  isValid: (string) => boolean,
+  setShouldValidate: (Function) => () => void,
+  setValue: (Function) => (string) => void,
+}
+
 
 export type User = {
   id: ?string,
-  email: ?string,
+  email: UserFormFieldAttribute,
   displayName: ?string,
-  firstName: ?string,
-  lastName: ?string,
+  firstName: UserFormFieldAttribute,
+  lastName: UserFormFieldAttribute,
   isTestUser: ?boolean,
   isPostDeploymentTestUser: boolean,
-  fullName?: string,
+  fullName: UserFormFieldAttribute,
   stateField?: string,
   gnmMarketing: boolean,
   isSignedIn: boolean,
 };
 
-
 // ----- Setup ----- //
 
 const initialState: User = {
   id: '',
-  email: '',
+  firstName: {
+    value: '',
+    shouldValidate: false,
+    isValid: (value: string) => !emptyInputField(value),
+    setShouldValidate: (dispatch: Function) => () => dispatch(setFirstNameShouldValidate()),
+    setValue: (dispatch: Function) => (value: string) => dispatch(setFirstName(value)),
```

Comment:
The Redux docs [strongly recommend](https://redux.js.org/faq/organizingstate#can-i-put-functions-promises-or-other-non-serializable-items-in-my-store-state) not keeping functions in the Redux store, as they are not serialisable. With the approach described above, `value` and `shouldValidate` are kept in the store, `isValid` is a helper function of type `(value: string) => boolean`, and `setShouldValidate` and `setValue` are actions that get mapped to the component.